protocol/str.go

```diff
@@ -0,0 +1,31 @@
+package protocol
+
+import (
+	"bytes"
+
+	"github.com/coniks-sys/coniks-go/crypto"
+	"github.com/coniks-sys/coniks-go/merkletree"
+)
+
+type DirSTR struct {
+	merkletree.SignedTreeRoot
+	Policies *Policies
+}
+
+func NewDirSTR(str *merkletree.SignedTreeRoot) *DirSTR {
+	return &DirSTR{
+		*str,
+		str.Ad.(*Policies),
+	}
+}
+
+// VerifyHashChain computes the hash of savedSTR's signature,
+// and compares it to the hash of previous STR included
+// in the issued STR. The hash chain is valid if
+// these two hash values are equal and consecutive.
+func (str *DirSTR) VerifyHashChain(savedSTR *DirSTR) bool {
```

Comment:
My point is I want to make `merkletree` be an independent and complete package and I think `merklettree.SignedTreeRoot` also needs a verifying method. Please feel free to discard this change :)